numalogic/udfs/tools.py

```diff
@@ -144,3 +146,104 @@ def _load_artifact(
                 },
             )
         return artifact, payload
+
+
+class TrainMsgDeduplicator:
+    """
+    TrainMsgDeduplicator class is used to deduplicate the train messages.
+    Args:
+        r_client: Redis client.
+    """
+
+    __slots__ = ("client", "_msg_read_ts", "_msg_train_ts")
+
+    def __init__(self, r_client: redis_client_t):
+        self.client = r_client
+        self._msg_read_ts: Optional[str] = None
+        self._msg_train_ts: Optional[str] = None
+
+    def __fetch_ts(self, key):
+        try:
+            data = self.client.hgetall(key)
+        except RedisError:
+            _LOGGER.exception("Problem  fetching ts information for the key: %s", key)
```

Comment:
What are we trying to do here after catching the error?

Reply:
Made it more clear now